statsmodels/stats/multitest.py

```diff
@@ -366,6 +373,91 @@ def fdrcorrection(pvals, alpha=0.05, method='indep', is_sorted=False):
     else:
         return reject, pvals_corrected
 
+def lfdrcorrection(pvals, pi_0_est=1., alpha=0.2, is_sorted=False):
+    '''
+    p-value correction for the local false discovery rate (lfdr).
+
+    Uses the support line (SL) procedure based on the Grenander estimator.
+
+    Parameters
+    ----------
+    pvals : array_like, 1d
+        List of p-values of the individual tests.
+    pi_0_est : float, optional
+        estimate of the null proportion. Defaults to conservative choice ``1.``.
+    alpha : float, optional
+        lfdr tolerance. Defaults to ``0.2``.
+    is_sorted : bool, optional
+        If False (default), the p-values will be sorted, but the corrected
+        p-values are in the original order. If True, then it assumed that the
+        p-values are already sorted in ascending order.
+
+    Returns
+    -------
+    rejected : ndarray, bool
+        True if a hypothesis is rejected, False if not
+    pvalue-corrected : ndarray
+        estimated lfdr values, i.e. adjusted p-values 
+
+    Examples
+    --------
+
+    >>> from statsmodels.stats.multitest import lfdrcorrection
+    >>> import numpy as np
+    >>> pvals = np.random.rand(30)
+    >>> rejected, lvals = lfdrcorrection(pvals)
+
+
+    References
+    ----------
+
+    U Grenander (1956). On the theory of mortality measurement: part II. 
+    Scandinavian Actuarial Journal, 39, 125-153.
+
+    B Efron, R Tibshirani, J D Storey, and V Tusher (2001). Empirical Bayes 
+    analysis of a microarray experiment. Journal of the American Statistical 
+    Association, 96:456, 1151-1160.
+
+    K Strimmer (2008). A unified approach to false discovery rate estimation. 
+    BMC Bioinformatics, 9, 1-14.
+
+    J A Soloff, D Xiang, and W Fithian (2024). The edge of discovery: 
+    Controlling the local false discovery rate at the margin. The Annals of 
+    Statistics, 52:2, 580-601.
+
+    Notes
+    -----
+
+    Assumes p-values are uniformly distributed under the null and have a
+    decreasing density under the alternative.
```

Comment:
I added independence as an assumption in the docstring since the main theory in the article assumes independence. Proposition 8 in the article allows for some dependence but it is a weaker result. Identically distributed is not necessary, however. Here is a result, mentioned in the discussion of the paper, to support this.

> Suppose $p_1,\ldots,p_m$ are independent random variables on $[0,1]$ such that, if the $i$-th null $H_i$ is true, the corresponding $p$-value is uniformly distributed $p_i\sim \text{Uniform}(0,1)$. For the SL $(\alpha)$ procedure, the probability that the last rejection is null is exactly $\frac{m_0}{m}\alpha$, where $m_0$ denotes the number of nulls.

Interestingly, this result doesn't require decreasing alternative densities, but it is only justified to control the probability that the _last rejection_ is null if the alternative densities are decreasing.